glog.go

```diff
@@ -2,218 +2,146 @@ package glog
 
 import (
 	"fmt"
+	"math"
+	"os"
 	"sync"
 
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
 )
 
+var maxLevel Level = math.MaxInt32
 var logger = log.NewNopLogger()
 var mu = sync.Mutex{}
 
+// SetLogger redirects glog logging to the given logger.
+// It must be called prior any call to glog.
 func SetLogger(l log.Logger) {
 	mu.Lock()
 	logger = l
 	mu.Unlock()
 }
 
+// ClampLevel clamps the leveled logging at the specified value.
+// It must be called prior any call to glog.
+func ClampLevel(l Level) {
+	mu.Lock()
+	maxLevel = l
+	mu.Unlock()
+}
+
 type Level int32
 
 type Verbose bool
 
-func V(level Level) Verbose { return true }
+func V(level Level) Verbose { return level <= maxLevel }
+
+func logMsg(l log.Logger, f string, msg string) {
+	l.Log("func", f, "msg", msg)
+}
```

Comment:
I'm not sure `logMsg` is worthwhile here. It's not reducing line count and it adds another stack frame that makes use of `log.Caller` from go-kit even less helpful. Note that in the sample prometheus log output every line has `caller=glog.go:40`. Would it be so terrible to replace each use of `logMsg` with something like this?
```go
level.Debug(logger).Log("func", "Verbose.Info", "msg", fmt.Sprint(args...)
```
That way the `caller` fields would at least be differentiated by which specific glog function was used.

Reply:
Sounds good to me!

Reply:
> Would it be so terrible to replace each use of logMsg with something like this?

Not at all. I'll send an update tomorrow. Thanks for the quick feedback. I discussed with @metalmatze offline but I'll send another PR updating the readme to explain how you can use it with Go modules (which is what Prometheus uses right now).